x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java

```diff
@@ -174,6 +174,7 @@ public class ClusterPrivilegeResolver {
     );
 
     private static final Set<String> MANAGE_SEARCH_APPLICATION_PATTERN = Set.of("cluster:admin/xpack/application/search_application/*");
+    private static final Set<String> MANAGE_SEARCH_CONNECTOR_PATTERN = Set.of("cluster:admin/xpack/connector/*");
```

Comment:
one side effect is that `manage_search_connector` will include the `read_connector_secret` and `write_connector_secret`, but that is the intention right?

https://github.com/elastic/elasticsearch/blob/6e031f58a2690df0b7b828120dbb129b4182161e/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilegeResolver.java#L368-L380

Reply:
It is a requirement that the user managing connectors has `write_connector_secrets`, so that API keys can be stored for the Connector (only relevant if the Connector is native). So that is okay to include.
Regarding `read_connector_secrets`, I don't think it's a requirement for users to have this. It should only be important for the Connector itself to read the API key.
If a user needs `read_connector_secrets` for some reason they could manually assign it to whichever role anyway, there is no restriction from doing this.

Reply:
Good spot! I see that it's possible to define `excludePatterns` in the [ActionClusterPrivilege](https://github.com/elastic/elasticsearch/blob/caf4bc1fbed1500a246471df2330736ff812247e/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ActionClusterPrivilege.java#L20) constructor.
```
     * Constructor for {@link ActionClusterPrivilege} that defines what cluster actions are accessible for the
     * user with this privilege after excluding the action patterns {@code excludedActionPatterns} from the allowed action patterns
     * {@code allowedActionPatterns}
 ```

 If I read the javadoc correctly it should allow us to match `cluster:admin/xpack/connector/*` while it's possible to exclude secrets-related privileges.
 
@navarone-feekery do you think the best way forward is to exclude all secret-related privileges from `manage_search_connector` to keep status quo? Right now API key generated used by connector service doesn't have this privilege right?

Reply:
> do you think the best way forward is to exclude all secret-related privileges from manage_search_connector to keep status quo?

I think having `write_connector_secrets` be separate for now is okay. It is already a [documented requirement for creating a native connector](https://www.elastic.co/guide/en/enterprise-search/current/native-connectors.html#native-connectors-index). It has a slightly different scope and we can always add it later if we expand secrets to do more things (like storing sensitive RCF values).

> Right now API key generated used by connector service doesn't have this privilege right?

Yes that's correct. Native connectors use the Enterprise Search system role to fetch the API key from connector secrets, then native connectors use that API key to index documents. The API key that a Connector uses itself doesn't need access to secrets.